drivers/storage/portworx/component/portworx_api.go

```diff
@@ -202,19 +202,31 @@ func (c *portworxAPI) createDaemonSet(
 
 	//  If px version is greater than 2.13 then update daemonset when csi-node-driver-registrar image changes if CSI is enabled
 	pxVersion := pxutil.GetPortworxVersion(cluster)
+	removeCSIRegistrar := false
+	addCSIRegistrar := false
 	checkCSIDriverRegistrarChange := false
 	if pxutil.IsCSIEnabled(cluster) && pxVersion.GreaterThanOrEqual(csiRegistrarAdditionPxVersion) && len(existingDaemonSet.Spec.Template.Spec.Containers) > 1 {
 		existingCsiDriverRegistrarImageName = existingDaemonSet.Spec.Template.Spec.Containers[1].Image
 		csiNodeRegistrarImageName = cluster.Status.DesiredImages.CSINodeDriverRegistrar
 		csiNodeRegistrarImageName = util.GetImageURN(cluster, csiNodeRegistrarImageName)
 		checkCSIDriverRegistrarChange = true
```

Comment:
since we have multiple if else statements now where we are computing same values like IsCsiEnabled multiple times, it might be better idea to store them before hand and use in if else statements

isCSIEnabled := pxutil.IsCSIEnabled(cluster)
isPxVersionCompatible := pxVersion.GreaterThanOrEqual(csiRegistrarAdditionPxVersion)
hasMultipleCsiContainers := len(existingDaemonSet.Spec.Template.Spec.Containers) > 1
isSingleCsiContainer := len(existingDaemonSet.Spec.Template.Spec.Containers) == 1